src/runtime/virtcontainers/qemu_s390x.go

```diff
@@ -66,6 +66,7 @@ func newQemuArch(config HypervisorConfig) (qemuArch, error) {
 			kernelParamsNonDebug: kernelParamsNonDebug,
 			kernelParamsDebug:    kernelParamsDebug,
 			kernelParams:         kernelParams,
+			legacySerial:         false,
```

Comment:
I guess this is the consequence of the `I'm not sure about s390x (as another device is used)` in the issue, right ?

Indeed, s390x is using a `virtio-serial-ccw` instead of `virtio-serial` but at the QEMU level, `virtio-serial` is an alias to `virtio-serial-ccw` AFAIK... ccw is _just_ the transport layer used on s390x, instead of PCI for other archs, but actual device is the same `virtio-serial`. In short, there are chances that s390x experiences the same absence of early boot logs and could benefit from this PR as well.

Maybe @BbolroC can shed some light here ?

Reply:
Hmm, maybe, also cc'ing @Jakob-Naucke

Reply:
Hi, unfortunately I did not have any chance to go over any PRs which @Jakob-Naucke made, but CI. I could make a comment to in June (my official start of the role for kata) because I am being involved in other projects which I had to finish by the end of this month. Or @hbrueckner could add some comments here.

Reply:
Adding this support in s390x can be done in a followup PR when technical details have been sorted out.

Reply:
I can't check super regularly but a quick chime in: I _think_ so, but I can't really test it